server/sonar-server/src/main/java/org/sonar/server/es/IndexingResult.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.es;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+public class IndexingResult {
+
+  // FIXME should be private
+  AtomicLong total = new AtomicLong(0L);
+  private long successes = 0L;
+
+  IndexingResult clear() {
+    total.set(0L);
+    successes = 0L;
+    return this;
+  }
+
+  void incrementRequests() {
+    total.incrementAndGet();
+  }
+
+  IndexingResult incrementSuccess() {
+    successes += 1;
+    return this;
+  }
+
+  public void add(IndexingResult other) {
+    total.addAndGet(other.total.get());
+    successes += other.successes;
+  }
+
+  public long getFailures() {
+    return total.get() - successes;
+  }
+
+  public long getTotal() {
+    return total.get();
+  }
+
+  public long getSuccess() {
+    return successes;
+  }
+
+  /**
+   * Get the failure ratio,
+   * if the total is 0, we always return 1 in order to break loop
+   * @see {@link RecoveryIndexer#recover()}
+   */
+  public double getFailureRatio() {
+    return total.get() == 0 ? 1 : (1.0d * getFailures()) / total.get();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add parentheses to make the operator precedence explicit. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS864)